src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Tree/Dependencies/Snapshot/TargetedDependenciesSnapshot.cs

```diff
@@ -110,7 +111,7 @@ void Remove(RemoveDependencyContext context, IDependencyModel dependencyModel)
                     }
                 }
 
-                dependencyById.Remove((dependencyModel.ProviderType, dependencyModel.Id));
+                dependencyById.Remove(dependencyModel.GetDependencyId());
```

Comment:
Would saving this value into a local remove an unnecessary allocation from this 2nd call?

Reply:
`DependencyId` is a value type so there's no extra allocation in this case. Any difference in the generated code will be minimal.

<!--

https://sharplab.io/#v2:EYLgdgpgLgZgHgGiiAhgZwLYB8ACAmARgFgAoHABgAIcCA6AJQFcwoBLDCWgYQHsMAHVgBsIAJwDKYgG6sAxhDQBuUqRwBmSqIgoAJjzBCAnpTRRRjWVEoARCPwhgdD2YYCSO0gG9SlX9Q00VACClJ6UAObQipQAvj5+6tQEVABCoRFRsfG+ibb2js5uOgAUgZQoCElUwACU6SEAvOXRaU3A0XEknaoBeDZ2Dk5gLl7Z/lWUIWGRUB1jiWVp05mdfuN5g4WlyeWVZbX1lE0oLUeU7VldKmQBBABs1H1coyRrC/fUACyUALIExRsCsNjE5+DUxt5Xms/AA5Yqg2gAcWggKGLncxRqNWUUOhlDhCORUFRhQxWOieMoY1WfnmtweOG+PzwAIGQJclFB4NxoTGaykKFElFYOjOhJRbLRRUxOMp+OKIuxfNhCp0Sp5NN8YwA2j9oAALHg6VwCITFPVQQ3G00AeX4bH0aFoMJ4rgMrDAHvCNQAunSkgzvnCScD3MK1elNVSee8HiH0aKifHpZbWGh+vkpZzJYU6g0AHyUSAAdwzm1DJVB5ZctCClSr7MMtBS6piQA=

-->